src/axom/primal/geometry/BezierPatch.hpp

```diff
@@ -976,25 +1048,70 @@ class BezierPatch
       return true;
     }
 
-    PlaneType the_plane = make_plane(m_controlPoints(0, 0),
-                                     m_controlPoints(0, ord_v),
-                                     m_controlPoints(ord_u, 0));
+    // Check that the four corners aren't coplanar
+    VectorType v1(m_controlPoints(0, 0), m_controlPoints(0, ord_v));
+    VectorType v2(m_controlPoints(0, 0), m_controlPoints(ord_u, 0));
+    VectorType v3(m_controlPoints(0, 0), m_controlPoints(ord_u, ord_v));
+    if(!axom::utilities::isNearlyEqual(
+         VectorType::scalar_triple_product(v1, v2, v3),
+         0.0,
+         tol))
+      return false;
+
+    // Find three points that produce a nonzero normal
+    Vector3D plane_normal = VectorType::cross_product(v1, v2);
+    if(axom::utilities::isNearlyEqual(plane_normal.norm(), 0.0, tol))
+      plane_normal = VectorType::cross_product(v1, v3);
+    if(axom::utilities::isNearlyEqual(plane_normal.norm(), 0.0, tol))
+      plane_normal = VectorType::cross_product(v2, v3);
+    plane_normal = plane_normal.unitVector();
 
     double sqDist = 0.0;
 
     // Check all control points for simplicity
     for(int p = 0; p <= ord_u && sqDist <= tol; ++p)
     {
-      for(int q = 0; q <= ord_v && sqDist <= tol; ++q)
+      for(int q = ((p == 0) ? 1 : 0); q <= ord_v && sqDist <= tol; ++q)
       {
-        double signedDist = the_plane.signedDistance(m_controlPoints(p, q));
+        double signedDist = VectorType::dot_product(
+          plane_normal,
+          Vector3D(m_controlPoints(0, 0), m_controlPoints(p, q)));
         sqDist += signedDist * signedDist;
       }
     }
 
     return (sqDist <= tol);
   }
 
+  /*!
+   * \brief Predicate to check if the Bezier patch is approximately planar-polygonal
+   *
+   * This function checks if the edges of a BezierPatch are approximately linear
```

Comment:
```suggestion
   * \brief Predicate to check if the patch can be approximated by a polygon
   *
   * This function checks if a BezierPatch lies in a plane
   *  and that the edged are linear up to tolerance `tol`
```

Is this any more clear? There's a part of me that wants to insist that the phrase "planar polygon" is a little redundant, since even in 3D space a polygon would be expected to lie in a plane.